packages/compass-aggregations/src/components/focus-mode/focus-mode-stage-preview.tsx

```diff
@@ -121,38 +172,63 @@ export const OutputPreview = (props: Omit<FocusModePreviewProps, 'title'>) => {
 
 export const FocusModeStageInput = connect(({
   focusMode: { stageIndex },
+  env,
   inputDocuments,
   pipelineBuilder: {
     stageEditor: {
       stages,
     }
   }
 }: RootState) => {
-  const previousStage = stages[stageIndex - 1];
-  return previousStage
-  ? {
+  if (stageIndex === -1) {
+    return null;
+  }
+
+  const previousStage = stages
+    .slice(0, stageIndex)
+    .filter(x => !x.disabled)
+    .reverse()[0];
+  if (!previousStage) {
+    return {
+      isLoading: inputDocuments.isLoading,
+      documents: inputDocuments.documents,
+    };
+  }
+
+  return {
     isLoading: previousStage.loading,
     documents: previousStage.previewDocs,
-  } : {
-    isLoading: inputDocuments.isLoading,
-    documents: inputDocuments.documents,
+    stageIndex: stageIndex - 1,
```

Comment:
I am not sure I follow why you need to do -1 here, wouldn't that be previous stage index and not a current one? If so, why are we keeping stage index that is off by one in the plugin state?

Reply:
This is the connector for input and its a stage before the current one and which's where the -1 is coming from. 
But, it has to be the index of `previousStage` not -1 as last stage can be disabled.

Reply:
Ahhhh, yes, this makes sense, thanks for explaining, I completely misread what this code is related to!